frontend/components/FieldMedia/FieldMediaEdit.jsx

```diff
@@ -137,103 +137,131 @@ export default class FieldImageEdit extends Component {
+                  size="small"
+                ><span>×</span></Button>
+              </div>
+            )
+          })
+        }
 
+        <div class={styles.upload}>
+          <div class={styles['upload-options']}>
+            <DropArea
+              draggingText={`Drop file${singleFile ? '' : 's'} here`}
+              onDrop={this.handleFileChange.bind(this)}
+            >
+              <div class={styles['upload-drop']}>
+                Drop file{singleFile ? '' : 's'} to upload
+              </div>
+            </DropArea>
+          </div>
+
+          <div class={styles.placeholder}>
             <Button
-              accent="destruct"
+              accent="neutral"
               size="small"
-              className={styles['remove-existing']}
-              onClick={this.handleRemoveFile.bind(this)}
-            >Delete</Button>
-          </div>          
-        }
+              href={href}
+            >Select existing {fieldLocalType.toLowerCase()}</Button>
+          </div>
 
-        {!values &&
-          <div>
-            <div class={styles['upload-options']}>
-              <DropArea
-                draggingText={`Drop image${singleFile ? '' : 's'} here`}
-                onDrop={this.handleFileChange.bind(this)}
-              >
-                <div class={styles['upload-drop']}>
-                  Drop file{singleFile ? '' : 's'} to upload
-                </div>
-              </DropArea>
-            </div>
-
-            <div class={styles.placeholder}>
-              <Button
-                accent="neutral"
-                size="small"
-                href={href}
-              >Select existing {fieldLocalType.toLowerCase()}</Button>
-            </div>
-
-            <div class={styles['upload-select']}>
-              <FileUpload
-                allowDrop={true}
-                accept={accept}
-                multiple={!singleFile}
-                onChange={this.handleFileChange.bind(this)}
-              />
-            </div>
+          <div class={styles['upload-select']}>
+            <FileUpload
+              accept={accept}
+              allowDrop={true}
+              multiple={!singleFile}
+              onChange={this.handleFileChange.bind(this)}
+            />
           </div>
-        }
+        </div>
       </Label>
     )
   }
+  
+  handleRemoveFile(fileName) {
+    const {name, onChange, schema, value} = this.props
+    const values = (value && !Array.isArray(value)) ? [value] : value
+    let newValues = values.filter((v) => v.fileName !== fileName)
 
-  getImageSrc(value) {
-    const {config} = this.props
-    const cdn = config ? config.cdn : null
-
-    if (!value) return null
+    if (newValues.length === 0) {
+      newValues = null
+    }
 
-    if (value._previewData) return value._previewData
+    if (typeof onChange === 'function') {
+      onChange.call(this, name, newValues)
+    }
+  }
 
-    if (value.url) return value.url
+  handleAddFiles(files) {
```

Comment:
We don't seem to be using this method anywhere and most of its logic seems to be repeated on line 288.